### PR TITLE
Permit underscore character as alternate separator

### PR DESCRIPTION
**Pull request by clemensv**
Addresses #932

### CODE REVIEW

File: cloudevents/bindings/amqp-protocol-binding.md [hunk 1]
```diff
@@ -169,14 +169,26 @@ time of the submission or revision.
...
+CloudEvent attributes MUST be prefixed with either "cloudEvents_" or 
+"cloudEvents:" for use in the application-properties section.
+
+The '\_' separator character SHOULD be preferred in the interest of 
+compatibility with JMS 2.0 clients and JMS message selectors where the
+':' separator is not permitted for property identifiers 
+(see section 3.8.1.1 of [JMS20]).
+
+CloudEvents AMQP consumers SHOULD understand the "cloudEvents" prefix with both, 
+the '\' and the ':' separators as permitted within the constraints of the client
```
Comment (duglin) [inline cloudevents/bindings/amqp-protocol-binding.md:181]: I think this needs to be `\\` (escaped). And next line too

File: cloudevents/bindings/amqp-protocol-binding.md [hunk 2]
```diff
@@ -169,14 +169,26 @@ time of the submission or revision.
...
+"cloudEvents:" for use in the application-properties section.
+
+The '\_' separator character SHOULD be preferred in the interest of 
+compatibility with JMS 2.0 clients and JMS message selectors where the
+':' separator is not permitted for property identifiers 
+(see section 3.8.1.1 of [JMS20]).
+
+CloudEvents AMQP consumers SHOULD understand the "cloudEvents" prefix with both, 
+the '\' and the ':' separators as permitted within the constraints of the client 
+model. JMS 2.0 AMQP consumers MUST understand the '\' separator; they cannot
```
Comment (duglin) [inline cloudevents/bindings/amqp-protocol-binding.md:182]: Missing `_`?

File: cloudevents/bindings/amqp-protocol-binding.md [hunk 3]
```diff
@@ -308,6 +318,7 @@ content-type: application/cloudevents+json; charset=utf-8
   Format
 - [OASIS-AMQP-1.0][oasis-amqp-1.0] OASIS Advanced Message Queuing Protocol
   (AMQP) Version 1.0
+ -[JMS20][JMS20] JSR-343 Java Message Service 2.0
```
Comment (duglin) [inline cloudevents/bindings/amqp-protocol-binding.md:321]: Swap the space and "-". The MD->HTML conversion looks funky:
image

File: cloudevents/bindings/amqp-protocol-binding.md [hunk 4]
```diff
@@ -169,14 +169,24 @@ time of the submission or revision.
...
-[application-properties][app-properties] section.
+CloudEvent attributes MUST be prefixed with either "cloudEvents_" or
+"cloudEvents:" for use in the application-properties section.
+
+The '\_' separator character SHOULD be preferred in the interest of
+compatibility with JMS 2.0 clients and JMS message selectors where the ':'
+separator is not permitted for property identifiers (see section 3.8.1.1 of
+[JMS2.0][JMS20]).
+
+CloudEvents AMQP consumers SHOULD understand the "cloudEvents" prefix with both,
```
Comment (jskeet) [inline cloudevents/bindings/amqp-protocol-binding.md:180]: Is the comma at the end of this line accidental?

**Comment by duglin** [comment]
Approved on the 8/25 call

**Pull request merged**